lldb/include/lldb/Symbol/Symtab.h

```diff
@@ -23,6 +23,8 @@ class Symtab {
 public:
   typedef std::vector<uint32_t> IndexCollection;
   typedef UniqueCStringMap<uint32_t> NameToIndexMap;
+  typedef std::map<lldb::addr_t, lldb_private::AddressClass>
+      FileAddressToAddressClassMap;
```

Comment:
You might want to use a llvm::DenseMap if possible. std::map uses 3 pointers for the red/black tree implementation and DenseMap is much more efficient I believe.

Reply:
This code depends on the map being sorted, which densemap isn't. According to the [llvm programmer's manual](https://llvm.org/docs/ProgrammersManual.html#map-like-containers-std-map-densemap-etc), the recommended data structure for a populate-then-query usage pattern is a sorted vector (though I would say that changing the data structure is not necessary for a patch like this).

Reply:
Yes, we binary search it later. I will at least add a comment to explain the type choice.